amp/WEB-INF/src/org/digijava/module/aim/action/CompareActivityVersions.java

```diff
@@ -365,11 +367,16 @@ public ActionForward compareAll(ActionMapping mapping, ActionForm form, HttpServ
                                     HttpServletResponse response) throws Exception {
 
         CompareActivityVersionsForm vForm = (CompareActivityVersionsForm) form;
+        HttpSession session = request.getSession();
+        TeamMember tm = (TeamMember) session.getAttribute("currentMember");
```

Comment:
Is better to check agains session atribute "ampAdmin" to check wheter is admin or not, also you should check if tm.getTeamHead() to avoid the page being accesed by non approvers/managers of the team

Reply:
I don't see why we need to check if user is manager. The user couldn't even access the audit trail page if not manager right?

Reply:
compareActivityVersions.do can be accessed directly, and if you know what parameters to send it would show your results even if not an admin. Its being extra precautious since its rare that anyone would try to hack the page, but technically its possible since compareActivityVersions.do is not secured to non admins since it needs to be accessed both via admin and audit trail